root/etc/e-smith/events/actions/nethserver-freepbx-fix-rsync-perms

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+#
+# Copyright (C) 2022 Nethesis S.r.l.
+# http://www.nethesis.it - support@nethesis.it
+# 
+# This script is part of NethServer.
+# 
+# NethServer is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License,
+# or any later version.
+# 
+# NethServer is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with NethServer.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# skip event name
+
+shift
+
+# check if engine used is rsync
+
+backup=$1
+[ "$(/sbin/e-smith/db backups gettype "${backup}")" == "rsync" ] || exit 0
+
+# if rsync then fix permissions
+
+if [[ -d "/var/lib/asterisk/" && $(stat -c %U:%G "/var/lib/asterisk/") != "asterisk:asterisk" ]]; then
+  /bin/chown asterisk:asterisk /var/lib/asterisk
```

Comment:
That path is not included because it's not involved in `backup-data`:
```
~]# cat /etc/backup-data.d/*.include | grep "lib/asterisk"
/var/lib/asterisk/moh
/var/lib/asterisk/sounds/en/custom
/var/lib/asterisk/sounds/it/custom
/var/lib/asterisk/sounds/nethcti
```
Its permissions and ownership should not be changing by the disaster recovery.
Shall we need to consider that path anyway?

Reply:
Very good, don't add it